src/jenkins_job_insight/main.py

```
@@ -3378,6 +3449,285 @@ def _serve_spa() -> HTMLResponse:
+@app.post("/api/admin/users/{username}/rotate-key")
+async def rotate_key_endpoint(request: Request, username: str) -> JSONResponse:
+    """Rotate an admin user's API key."""
+    _require_admin(request)
+    try:
+        body_bytes = await request.body()
+        if body_bytes and body_bytes.strip():
+            body = await request.json()
+            custom_key = body.get("new_key")
+        else:
+            custom_key = None
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)